### PR TITLE
[gui] refactor visualisation preset dialog

### PR DESCRIPTION
**Pull request by tamland**
Fixes cancel button regression after #8417 as detailed here: http://forum.kodi.tv/showthread.php?tid=261387&pid=2258102

@un1versal

### CODE REVIEW

**Comment by BigNoid** [comment]
Much better than #8863. Will close that one.

**Comment by tamland** [comment]
Oh, didn't even see that one. Can't believe nobody pointed out how hacky this preset dialog is:)

**Comment by MartijnKaijser** [comment]
@tamland i'm sure they knew and staid far, far away from it :)

**Comment by un1versal** [comment]
works, thx a lot :)

screenshot083

File: xbmc/dialogs/GUIDialogSelect.cpp [hunk 1]
```diff
@@ -109,8 +111,7 @@ bool CGUIDialogSelect::OnMessage(CGUIMessage& message)
               for (int i = 0 ; i < m_vecList->Size() ; i++)
                 m_vecList->Get(i)->Select(false);
               item->Select(true);
-              m_bConfirmed = true;
-              Close();
+              OnSelect(iSelected);
```
Comment (Razzeee) [inline xbmc/dialogs/GUIDialogSelect.cpp]: Why are you moving this? I can't find another way it's called.

**Comment by tamland** [comment]
jenkins build this please

**Pull request merged**